app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -587,6 +585,493 @@ class FeatureFlagsFragmentTest {
     }
   }
 
+  @Test
+  fun testFeatureFlagsFragment_withOverriddenFlag_resetButtonIsVisible() {
+    executeInPreviousAppInstance { component ->
+      addTestOverriddenFeatureFlagToDatabase(component, true)
+      component.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.feature_flags_recycler_view,
+          position = 0,
+          targetViewId = R.id.reset_button
+        )
+      ).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_withNoRemoteAndwithOverride_clickResetButton_resetsFlagToDefault() {
+    executeInPreviousAppInstance { component ->
+      addTestOverriddenFeatureFlagToDatabase(component, true)
+      component.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.feature_flags_recycler_view,
+          position = 0,
+          targetViewId = R.id.reset_button
+        )
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.feature_flags_recycler_view,
+          position = 1,
+          targetViewId = R.id.feature_flag_switch
+        )
+      ).check(matches(not(isChecked())))
+    }
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_withNoRemoteAndwithOverride_onResetshowsNeverSyncedMessage() {
```

Comment:
Here and below, the last phrase to only focus on the `_result`.
```suggestion
  fun testFeatureFlagsFragment_withOverrideAndNoRemote_onReset_showsNeverSyncedMessage() {
```

Reply:
updated